webrtc.html

```diff
@@ -1272,6 +1272,11 @@ <h4>Set the RTCSessionDescription</h4>
                     <p>If <var>connection</var>.<a>[[\IsClosed]]</a> is
                     <code>true</code>, then abort these steps.</p>
                   </li>
+                  <li>
+                    <p>If the {{RTCPeerConnection/addTrack()}} method was called
+                    during the process to apply <var>description</var>, then
+                    abort these steps and start the process over.</p>
```

Comment:
Would aborting the processes and starting over implicitly involve rolling back some JSEP layer actions, or do you imagine "starting over" is always "safe"?

For example, if the first attempt to SRD() didn't have a transceiver to associate with then a transceiver will be proposed to be created. But when starting over due to simultaneous addTrack(), it would discover that there is a transceiver to associate with. Ergo, the proposed transceiver that was never surfaced to JS is thrown away.

It may be implementation-specific how realized this proposed transceiver is: if an object is created or if there is just a struct saying that there should be a transceiver with such and such mids, but no such transceiver exists. It is clear that the intent is that there will be no transceiver SRD-created transceiver as far as JS is concerned. Just raising this point to make sure we don't miss some edge case complexity...
- Should we clarify that starting the process over means that any side-effects caused by the previous processes must be un-done?
- Are there any other side-effects to applying the process twice? For example, could transports get created in the first round and then re-created in the second?

I'm hoping the answer is everything's cool, just wanted a second pair of eyes on this. WDYT @jan-ivar ?

Reply:
JSEP defines an all-or-nothing SetRemoteDescription and all-or-nothing rollback, but it does not define "starting over" during surfacing steps, so I imagine JSEP would have fully completed the SetRemoteDescription and then needs to change its mind somehow - but this may dig into our different views on webrtc-pc vs JSEP layers. I'd like both views to be valid interpretations for the sake of not having to re-implement things in either browser.

Reply:
Do we dare to add a class="note" that says "After the processes of starting over is complete, the resulting set of transceivers and their associations should be the same as if addTrack() was performed first and setRemoteDescription() was performed second"?

Reply:
> Would aborting the processes and starting over implicitly involve rolling back some JSEP layer actions, or do you imagine "starting over" is always "safe"?

@henbos I don't think there are side effects, except maybe a skip in mid sequencing which seems tolerable? I'd like @docfaraday to weigh in to be sure. We're not "aborting" the JSEP process, only the subsequent queued success task, which doesn't contain any transitional conditionals (coining it) AFAICT, so I think it's safe since we haven't exposed anything to JS yet, and we will after re-processing. Repeated SRD's must already be allowed according to:
![state diagram](https://w3c.github.io/webrtc-pc/images/peerstates.svg)

> could transports get created in the first round and then re-created in the second?

JS-exposed transports haven't been created yet (they're created in the queued task).

I've updated the prose a bit to clarify intent. PTAL